check-required-lite/README.md

```diff
@@ -0,0 +1,33 @@
+# check-required-lite
+
+A GitHub Action which checks if the given steps have completed successfully.
+
+By contrast with [check-required](https://github.com/G-Research/common-actions/tree/d8575b4a7ce45a5735363734f4bcf640e46aee1b/check-required),
+this action does not require a GitHub app.
+
+You must call the action as follows.
+The `if` clause is essential, as is the `needs-context`.
+The only part of this invocation which you vary is the contents of the `needs:` list.
+
+```yaml
+all-required-checks-complete:
+  needs: [some-previous-step, another-step]
+  if: ${{ always() }}
+  runs-on: ubuntu-latest
+  steps:
+    - uses: G-Research/common-actions/check-required-lite@main
+      with:
+        needs-context: ${{ toJSON(needs) }}
```

Comment:
is there a specific reason to use this instead of the `needs` context that is already provided to the action?
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#needs-context

Reply:
Yeah, I couldn't find any evidence that a `needs` context *is* provided to the action. Concretely, [deleting](https://github.com/Smaug123/all-required-checks-complete-action/pull/2/commits/7f00e93876ff088216da11c52e3f83a13ca5bf55) that context block [fails](https://github.com/Smaug123/WoofWare.DotnetRuntimeLocator/actions/runs/10456146766/job/28952877684?pr=32) with an error indicating that the `INPUT_NEEDS` env var is not populated.

Reply:
Indeed, even this fails:

```yaml
inputs:
  needs-context:
    description: 'JSON string of the needs context, which you should give as toJSON(needs)'
    required: false
    default: ${{ toJSON(needs) }}
```
> Error: Smaug123/all-required-checks-complete-action/c8f09fcdca5f03fc574af26064ab80982437466b/action.yaml (Line: 11, Col: 14): Unrecognized named-value: 'needs'. Located at position 8 within expression: toJSON(needs)